src/PrestaShopBundle/Translation/Provider/TranslationsCatalogueProvider.php

```diff
@@ -0,0 +1,266 @@
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Translation\Provider;
+
+use PrestaShop\PrestaShop\Core\Exception\FileNotFoundException;
+use PrestaShopBundle\Exception\NotImplementedException;
+use PrestaShopBundle\Translation\Provider\Factory\ProviderFactory;
+use PrestaShopBundle\Translation\Provider\Type\TypeInterface;
+use Symfony\Component\Translation\MessageCatalogue;
+use Symfony\Component\Translation\MessageCatalogueInterface;
+
+/**
+ * Retrieves combined and formatted catalogues depending on the provider type.
+ */
+class TranslationsCatalogueProvider
+{
+    /**
+     * @var ProviderFactory
+     */
+    private $providerFactory;
+
+    public function __construct(ProviderFactory $providerFactory)
+    {
+        $this->providerFactory = $providerFactory;
+    }
+
+    /**
+     * @param TypeInterface $providerType
+     * @param string $locale
+     * @param string $domain
+     * @param array $search
+     *
+     * @return array
+     *
+     * @throws FileNotFoundException|NotImplementedException
+     */
+    public function getDomainCatalogue(
+        TypeInterface $providerType,
+        string $locale,
+        string $domain,
+        array $search = []
+    ): array {
+        $provider = $this->providerFactory->build($providerType);
+
+        $defaultCatalogue = $provider->getDefaultCatalogue($locale);
+        if (null === $defaultCatalogue) {
+            return [];
+        }
+        $defaultCatalogue = $this->filterCatalogue($defaultCatalogue, $locale, $domain)->all($domain);
+
+        try {
+            $fileTranslatedCatalogue = $provider->getFileTranslatedCatalogue($locale);
+            $fileTranslatedCatalogue = (null !== $fileTranslatedCatalogue) ? $this->filterCatalogue($fileTranslatedCatalogue, $locale, $domain)->all($domain) : [];
+        } catch (FileNotFoundException $exception) {
+            $fileTranslatedCatalogue = [];
+        }
+
+        $userTranslatedCatalogue = $provider->getUserTranslatedCatalogue($locale);
+        $userTranslatedCatalogue = (null !== $userTranslatedCatalogue) ? $this->filterCatalogue($userTranslatedCatalogue, $locale, $domain)->all($domain) : [];
```

Comment:
Just to be sure, that's because `$provider->getUserTranslatedCatalogue` cannot throw exceptions that this part is not inside a `try catch` like just before?

Reply:
An exception is not expected here. If there is one, it can be related to DB access